### ISSUE TITLE
Conversion from java.net.URI to EMF URI drops encoding

### ISSUE THREAD

**Issue by mark-christiaens** [eclipse/xtext-core#1163]
In `UriExtensions.toEmfUri` I see

```
/**
 * returns a URI with empty authority, if absent and has file scheme.
 */
def URI toEmfUri(java.net.URI netUri) {
	val decoded = netUri.toDecodedString
	val uri = URI.createURI(decoded);
	val result = uri.withEmptyAuthority
	result
}
```

So the `uri` is created using a decoded string.  However, the JavaDoc of `URI.createURI` says:

```
...  This method also does not perform encoding of invalid characters, so it 
should only be used when the URI string is known to have already been 
encoded, so as to avoid double encoding.
```
Clearly, the URI string is _not_ encoded in `toEmfUri`.

**Comment by cdietrich** [comment]
am not sure if this can easily be fixed. see org.eclipse.xtext.ide.tests.server.UriExtensionsTest too

**Comment by cdietrich** [comment]
but there are situations where it crashes already now

**Comment by cdietrich** [comment]
just fixing it seems no to work. there are still places with maybe double en/decoding

**Comment by cdietrich** [comment]
Fixed in 2.19
